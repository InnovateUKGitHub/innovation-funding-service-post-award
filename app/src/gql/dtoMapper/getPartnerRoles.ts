/**
 * simply maps out partner roles to get rid of the annoying readonly tags
 */
export const getPartnerRoles = (
  rolesNode: Readonly<{ partnerRoles?: Readonly<Readonly<SfPartnerRoles>[]> } | null>,
) => {
  return (
    rolesNode?.partnerRoles?.map(x => ({
      isFc: x?.isFc ?? false,
      isPm: x?.isPm ?? false,
      isMo: x?.isMo ?? false,
      partnerId: x?.partnerId ?? "unknown partner id",
    })) ?? []
  );
};
