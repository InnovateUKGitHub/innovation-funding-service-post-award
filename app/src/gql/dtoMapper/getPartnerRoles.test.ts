import { getPartnerRoles } from "./getPartnerRoles";

describe("getPartnerRoles", () => {
  it("should map from the partnerRoles object to a list of partner roles", () => {
    const node = {
      partnerRoles: [
        {
          isFc: true,
          isPm: false,
          isMo: false,
          partnerId: "1",
        },
        {
          isFc: false,
          isPm: true,
          isMo: false,
          partnerId: "2",
        },
        {
          isFc: false,
          isPm: false,
          isMo: true,
          partnerId: "3",
        },
      ],
    };

    expect(getPartnerRoles(node)).toMatchSnapshot();
  });
});
