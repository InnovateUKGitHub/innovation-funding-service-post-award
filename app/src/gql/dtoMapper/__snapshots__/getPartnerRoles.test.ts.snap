// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPartnerRoles should map from the partnerRoles object to a list of partner roles 1`] = `
Array [
  Object {
    "isFc": true,
    "isMo": false,
    "isPm": false,
    "partnerId": "1",
  },
  Object {
    "isFc": false,
    "isMo": false,
    "isPm": true,
    "partnerId": "2",
  },
  Object {
    "isFc": false,
    "isMo": true,
    "isPm": false,
    "partnerId": "3",
  },
]
`;
